kinto_client/__init__.py

```diff
@@ -117,265 +110,113 @@ def request(self, method, url, data=None, permissions=None, **kwargs):
+        self._bucket_name = bucket
+        self._collection_name = collection
+        self.endpoints = Endpoints()
+
+    def _get_endpoint(self, name, bucket=None, collection=None, id=None):
+        kwargs = {
+            'bucket': bucket or self._bucket_name,
+            'collection': collection or self._collection_name,
+            'id': id
+        }
+        return self.endpoints.get(name, **kwargs)
+
+    # Buckets
+
+    def update_bucket(self, bucket=None, permissions=None):
+        endpoint = self._get_endpoint('bucket', bucket)
+        resp, _ = self.session.request('put', endpoint,
+                                       permissions=permissions)
+        return resp
+
+    def delete_bucket(self, bucket=None):
+        endpoint = self._get_endpoint('bucket', bucket)
+        resp, _ = self.session.request('delete', endpoint)
         return resp['data']
 
-    def save(self):
-        self.session.request('patch', self.uri, data=self.data,
-                             permissions=self.permissions)
-
-    def delete(self):
-        resp, _ = self.session.request('delete', self.uri)
-        return resp['data']
+    def get_bucket(self, bucket=None):
+        endpoint = self._get_endpoint('bucket', bucket)
+        try:
+            resp, _ = self.session.request('get', endpoint)
+        except KintoException as e:
+            exception = BucketNotFound(bucket or self._bucket_name)
+            exception.response = e.response
+            exception.request = e.request
+            raise exception
+        return resp
 
+    def create_bucket(self, *args, **kwargs):
+        # Alias to update bucket.
+        return self.update_bucket(*args, **kwargs)
 
-# XXX Refactor the permissions to use dicts + defaultdicts rather
-# than attributes.
-class Permissions(object):
-    """Handles the permissions as sets"""
-    def __init__(self, object, permissions=None):
-        objects = OBJECTS_PERMISSIONS.keys()
-        if object not in objects:
-            msg = 'object should be one of %s' % ','.join(objects)
-            raise AttributeError(msg)
+    # Collections
 
-        if permissions is None:
-            permissions = {}
+    def get_collections(self, bucket=None):
+        endpoint = self._get_endpoint('collections', bucket)
+        return self.session.request('get', endpoint)
 
-        self.object = object
-        self.permissions = permissions
+    def create_collection(self, collection=None, bucket=None,
+                          permissions=None):
+        endpoint = self._get_endpoint('collection', bucket, collection)
+        # XXX Add permissions
+        return self.session.request('put', endpoint, permissions=permissions)
 
-        for permission_type in OBJECTS_PERMISSIONS[object]:
-            attr = permission_type.replace(':', '_')
-            setattr(self, attr, permissions.get(permission_type, []))
+    def update_collection(self, *args, **kwargs):
```

Comment:
ditto: I am not sure if Python is capable of handling: ``create_collection = update_collection`` at a class level (also I find it more readable to have ``create_collection = update_collection`` than the other way around.